### PR TITLE
refactor: core `app` module

### LINKED ISSUE #3221: [Task]: Refactor core `app` module
### Task description

The `app*.ts` files need to be refactored into a dedicate `app` module. 

### Requirements

- [x] Move constants, enums, interfaces, stores, utils, etc. over to `app`
- [x] Improve settings and updater types
- [ ] ~~Change notion of "Platform" to "app"~~

### Acceptance criteria

Meets the above requirements.

### Creation checklist

- [X] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary
There is app logic in many different places throughout our code, and it could all benefit from having a dedicate `app` module.

### Changelog
```
- Create core "app" module
- Refactor logic for the general app, the settings, and the updater
```

## Relevant Issues
Closes #3221 

## Type of Change
Please select any type below that is applicable to your changes, and delete those that are not.
- [ ] __Breaking__ - any change that would cause existing functionality to not work as expected
- [x] __Chore__ - refactoring, build scripts or anything else that isn't user-facing
- [ ] __Docs__ - changes to the documentation
- [ ] __Fix__ - a change which fixes an issue
- [ ] __New__ - a change which implements a new feature
- [ ] __Update__ - a change which updates existing functionality

### Platforms
- __Desktop__
	- [ ] MacOS
	- [x] Linux
	- [ ] Windows
- __Mobile__
	- [ ] iOS
	- [ ] Android

### Instructions
Download and build app, and ensure that there are no visible changes to general functionality.

## Checklist
- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or modified tests that prove my changes work as intended
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have verified that my latest changes pass CI workflows for testing and linting
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/routes/setup/Welcome.svelte [hunk 1]
```diff
@@ -1,10 +1,8 @@
...
     import { appRouter } from '@core/router'
     import { Animation, Button, Dropdown, Logo, OnboardingLayout, Text } from 'shared/components'
-    import { mobile } from 'shared/lib/app'
-    import { appSettings } from 'shared/lib/appSettings'
-    import { SUPPORTED_LOCALES, setLanguage, _ } from '@core/i18n'
-    import { Locale } from '@core/i18n'
+    import { appSettings, mobile } from '@core/app'
+    import { SUPPORTED_LOCALES, setLanguage, _, Locale } from '@core/i18n'
 
     export let locale: Locale
```
Comment (Tuditi) [inline packages/shared/routes/setup/Welcome.svelte:7]: Let's remove locale here

File: packages/shared/routes/setup/legal/Legal.svelte [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+        TERMS_OF_SERVICE_VERSION,
+        PRIVACY_POLICY_VERSION,
+    } from '@core/app'
     import { Locale } from '@core/i18n'
     import Content from './Content.svelte'
-    import { lastAcceptedTos, lastAcceptedPrivacyPolicy } from 'shared/lib/appSettings'
-    import { TOS_VERSION, PRIVACY_POLICY_VERSION } from 'shared/lib/app'
     import { appRouter } from '@core/router'
 
     export let locale: Locale
```
Comment (Tuditi) [inline packages/shared/routes/setup/legal/Legal.svelte:14]: Same

**Review by Tuditi — CHANGES_REQUESTED** [review]
Code looks good, but I have a minor concern with regard to having multiple levels of folders within a core component. Specifically with `core/app/actions/app-update`. I don't see why the extra folder under `actions/` is warranted, since it only contains three functions. For me this is a bit of an overkill. I also couldn't test, since there is no synced node available atm.

File: packages/shared/lib/core/app/stores/app-stage.store.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { writable } from 'svelte/store'
+
+import { AppStage } from '../enums'
+
+/**
+ * Beta mode
+ */
```
[the remark this answers is not in the document]
Comment (maxwellmattryan) [reply packages/shared/lib/core/app/stores/app-stage.store.ts:7]: Yeah this comment is pretty useless. I'll be looking at generating spec pages for different modules based on TSDoc comments.

File: packages/shared/lib/core/app/actions/downloadAppUpdate.ts [hunk 4]
```diff
@@ -9,61 +9,36 @@ import {
...
-
-    const notificationId = showAppNotification(downloadingNotification)
+/**
+ * Initializes the download for an application update.
+ */
+export function downloadAppUpdate(): void {
+    resetAppUpdateStores()
 
-    progressSubscription = appUpdateProgress.subscribe((progress) => {
+    const progressSubscription: Unsubscriber = appUpdateProgress.subscribe((progress) => {
```
Comment (Tuditi) [inline packages/shared/lib/core/app/actions/downloadAppUpdate.ts:37]: I like your solution here, but you don't need to type the variables here. `subscribe()` can only return an `Unsubscriber`, so the type will be automatically inferred.

File: packages/shared/lib/core/app/actions/downloadAppUpdate.ts [hunk 5]
```diff
@@ -123,5 +98,33 @@ export function downloadAppUpdate(): void {
         }
     })
 
+    const cleanup = () => {
```
Comment (Tuditi) [inline packages/shared/lib/core/app/actions/downloadAppUpdate.ts:101]: Let's make this a regular function as well

**Review by Tuditi — APPROVED** [review]
Code looks good, I'll still need to test it in the app.

File: packages/shared/lib/core/app/actions/downloadAppUpdate.ts (hunk 5 above)
Comment (maxwellmattryan) [reply packages/shared/lib/core/app/actions/downloadAppUpdate.ts:101]: Yes, I think this makes sense. The function is used in the callback for the unsubscribers themselves, which seems to be / work okay although it makes it a bit weird to extract it out.

**Pull request merged**